### PR TITLE
Remove deprecated TensorMechanics syntax in remaining test inputs

### PR DESCRIPTION
**Pull request by bwspenc**
ref idaholab/moose#25584

### CODE REVIEW

**Comment by bwspenc** [comment]
Note that there were a number of tests that were mostly set up as small strain, but they still ran the kernels on the displaced mesh. I switched them over to actual small strain conditions, and had to regold the results because of that. I checked them, and they had relative differences on the order of 1e-3, and the new solutions are spot-on to the exact analytical values while the old ones were a little off in many cases.

File: test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_excluding_confinement_effect_out.csv
Comment (GiudGiud) [inline test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_excluding_confinement_effect_out.csv:1]: results changes are bigger

File: test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_free_out.csv
Comment (GiudGiud) [inline test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_free_out.csv:1]: even bigger here

File: test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_excluding_confinement_effect_out.csv
Comment (bwspenc) [reply test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_excluding_confinement_effect_out.csv:1]: They should nominally be zero. The new results are closer to zero, so they are better.

File: test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_free_out.csv
Comment (bwspenc) [reply test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_free_out.csv:1]: Oh, I just realized the comments are about the files below. This one looks better than the original, so I think it's good.

**Review by GiudGiud — COMMENTED** [review]
inputs changes look fine to me

File: test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_excluding_confinement_effect_out.csv
Comment (bwspenc) [reply test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_excluding_confinement_effect_out.csv:1]: Scratch what I said above (I was looking at the file above). These are a little higher than before, which isn't what we want. I'm looking into this one.

Comment (bwspenc) [reply test/tests/concrete_expansion_microcracking/gold/concrete_expansion_microcracking_excluding_confinement_effect_out.csv:1]: I tightened up the solve convergence tolerance, which drops the stresses all down to the 1e-10 range, which is what we want. I think this is in good shape now.

**Review by GiudGiud — APPROVED** [review]

**Review by dschwen — APPROVED** [review]

**Pull request merged**